packages/appsync-modelgen-plugin/src/utils/process-connections-v2.ts

```diff
@@ -0,0 +1,103 @@
+import { CodeGenField, CodeGenFieldDirective, CodeGenModel, CodeGenModelMap } from '../visitors/appsync-visitor';
+import {
+  CodeGenFieldConnection, DEFAULT_HASH_KEY_FIELD, flattenFieldDirectives,
+  getDirective,
+  makeConnectionAttributeName,
+} from './process-connections';
+import { processHasOneConnection } from './process-has-one';
+import { processBelongsToConnection } from './process-belongs-to';
+import { processHasManyConnection } from './process-has-many';
+
+// TODO: This file holds several references to utility functions in the v1 process connections file, those functions need to go here before that file is removed
+
+export function getConnectedFieldV2(field: CodeGenField, model: CodeGenModel, connectedModel: CodeGenModel, directiveName: string): CodeGenField {
+  const connectionInfo = getDirective(field)(directiveName);
+  if (!connectionInfo) {
+    throw new Error(`The ${field.name} on model ${model.name} is not connected`);
+  }
+
+  const connectionName = connectionInfo.arguments.name;
```

Comment:
According to the design, none of the three directives use `name` argument

Reply:
Got it, cutting that part out